// $ cd ./server
// $ python -m grpc_tools.protoc -I../protos --python_out=. --grpc_python_out=. ../protos/truco.proto

syntax = "proto3";
package trucopackage;
// import "google/protobuf/empty.proto";

service Truco{
    rpc showTables(Query) returns (stream Table);
    rpc createNewTable(Query) returns (QueryReply) {};
    rpc joinTable(Query) returns (QueryReply) {};
    rpc exitTable(Query) returns (QueryReply) {};
}

message Table {
    string name = 1;
    repeated string team_A = 2;
    repeated string team_B = 3;
    repeated int32 scoreboard = 4;
    string init = 5;
}

message Query {
    string tablename = 1;
    string nickname = 2;
    string team = 3;
}

message QueryReply {
    bool status = 1;
    string cmdQueue = 2;
    string cliQueue = 3;
}